jwst/resample/resample_step.py

```diff
@@ -170,12 +171,20 @@ def _check_list_pars(vals, name, min_vals=None):
             raise ValueError(f"Both '{name}' values must be either None or not None.")
 
     @staticmethod
-    def _load_custom_wcs(asdf_wcs_file, output_shape):
-        if not asdf_wcs_file:
+    def _load_custom_wcs(input_wcs_file, output_shape):
+        if not input_wcs_file:
             return None
 
-        with asdf.open(asdf_wcs_file) as af:
-            wcs = deepcopy(af.tree["wcs"])
+        try:
+            # File is ASDF file with a top-level wcs in the tree
+            with asdf.open(input_wcs_file) as af:
+                wcs = deepcopy(af.tree["wcs"])
+        except (KeyError, ValueError):
+            # File is ImageModel or SlitModel with the wcs under meta
+            with datamodels.open(input_wcs_file) as model:
+                wcs = deepcopy(model.meta.wcs)
+                if not output_shape:
+                    output_shape = model.data.shape[::-1]
```

Comment:
If I'm reading this correctly, it means that if an external reference file is provided the data shape takes precedence to the bounding box of the WCS (assuming it exists too). Is this the correct decision? It seems to me the bounding_box should take precedence.

Reply:
Not part of this PR but the line below (line 189) seems wrong, especially when `wcs is None`